operator/builtin/transformer/copy/copy_test.go

```diff
@@ -0,0 +1,223 @@
+						"nestedkey": "nestedval",
+					},
+					"key2": "val",
+				}
+				return e
+			},
+		},
+		{
+			"nested_to_body",
+			false,
+			func() *CopyOperatorConfig {
+				cfg := defaultCfg()
+				cfg.From = entry.NewBodyField("nested", "nestedkey")
+				cfg.To = entry.NewBodyField("key2")
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key": "val",
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+					"key2": "nestedval",
+				}
+				return e
+			},
+		},
+		{
+			"body_to_nested",
+			false,
+			func() *CopyOperatorConfig {
+				cfg := defaultCfg()
+				cfg.From = entry.NewBodyField("key")
+				cfg.To = entry.NewBodyField("nested", "key2")
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key": "val",
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+						"key2":      "val",
+					},
+				}
+				return e
+			},
+		},
+		{
+			"body_to_attribute",
+			false,
+			func() *CopyOperatorConfig {
+				cfg := defaultCfg()
+				cfg.From = entry.NewBodyField("key")
+				cfg.To = entry.NewAttributeField("key2")
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key": "val",
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+				}
+				e.Attributes = map[string]string{"key2": "val"}
+				return e
+			},
+		},
+		{
+			"attribute_to_body",
+			false,
+			func() *CopyOperatorConfig {
+				cfg := defaultCfg()
+				cfg.From = entry.NewAttributeField("key")
+				cfg.To = entry.NewBodyField("key2")
+				return cfg
+			}(),
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Attributes = map[string]string{"key": "val"}
+				return e
+			},
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key": "val",
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+					"key2": "val",
+				}
+				e.Attributes = map[string]string{"key": "val"}
+				return e
+			},
+		},
+		{
+			"attribute_to_resource",
+			false,
+			func() *CopyOperatorConfig {
+				cfg := defaultCfg()
+				cfg.From = entry.NewAttributeField("key")
+				cfg.To = entry.NewResourceField("key2")
+				return cfg
+			}(),
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Attributes = map[string]string{"key": "val"}
+				return e
+			},
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Attributes = map[string]string{"key": "val"}
+				e.Resource = map[string]string{"key2": "val"}
+				return e
+			},
+		},
+		{
+			"invalid_key",
+			true,
+			func() *CopyOperatorConfig {
+				cfg := defaultCfg()
+				cfg.From = entry.NewAttributeField("nonexistentkey")
+				cfg.To = entry.NewResourceField("key2")
+				return cfg
+			}(),
+			newTestEntry,
+			nil,
+		},
```

Comment:
Let's have some more tests:
- Overwrite a value
- Error when copying non-string to attributes/resource